### ISSUE TITLE
Performance: full hidden state downloaded to extract single row for logits

### ISSUE THREAD

**Issue by anibjoshi** [strata-ai-labs/strata-inference#69]
## Context

In `project_to_logits()` (`src/engine/generate.rs:374-393`), after each forward pass the **entire hidden state tensor** is downloaded from GPU to CPU, even though only a single row is needed for logit projection.

```rust
fn project_to_logits(&self, hidden: &DeviceTensor, row: usize) -> Vec<f32> {
    let hidden_host = self.backend.download(hidden);  // Downloads ALL rows
    let hidden_data = hidden_host.as_f32();
    let hidden_size = self.config.hidden_size;
    let row_data = &hidden_data[row * hidden_size..row * hidden_size + hidden_size];
    // ... then uploads row_data back for matmul
}
```

## Impact

- For a 7B model (hidden_size=4096), this transfers ~16KB per decode step — tolerable
- For prefill with seq_len=2048, this transfers ~8MB when only 1 row (~16KB) is needed
- Main cost is the GPU synchronization barrier, not bandwidth

## Proposed Fix

Add a `backend.extract_row(tensor, row_index)` operation that returns a single-row DeviceTensor without leaving the device. The logit projection matmul can then run entirely on-device.

```rust
fn project_to_logits(&self, hidden: &DeviceTensor, row: usize) -> Vec<f32> {
    let hidden_row = self.backend.extract_row(hidden, row);
    let logits = self.backend.matmul_transpose(&hidden_row, proj_weight);
    self.backend.download(&logits).as_f32().to_vec()
}
```

## Complexity

Low — requires one new backend operation (row extraction / slice) on CPU, Metal, and CUDA.
